### PR TITLE
Make heatmap colors configurable

### PR DESCRIPTION
**Pull request by jolheiser**
This allows all heatmap colors to be configured (even custom themes).

### CODE REVIEW

**Comment by jolheiser** [comment]
I was going to allow users to use `rgb` as well, but that would mean hex-codes would need the `#` prefix, which makes it a comment in INI format. 🙁 

It could still be done, but it would increase complexity a little.

**Comment by 6543** [comment]
@jolheiser hex colorbcode is fine as long as it is written somehow in the docs

**Comment by 6543** [comment]
Cant test itvat the moment boot it looks good!

**Review by kolaente — CHANGES_REQUESTED** [review]
Please also remove the colors in the css and update the docs, to make it clear where to change these.

**Comment by jolheiser** [comment]
> Please also remove the colors in the css and update the docs, to make it clear where to change these.

The colors are still needed in CSS, as those would be the fallback.  
And I did update the customization docs, was there another place to change it?

**Comment by kolaente** [comment]
> The colors are still needed in CSS, as those would be the fallback.

I would prefer having that only as default option in the config

> I didn't add anything to the sample ini since these are all optional and the keys change (somewhat) dynamically based on the theme names.

But wouldn't it make sense to have defaults for the light and dark theme?

**Comment by jolheiser** [comment]
If the colors are completely converted to config values, we lose some of the niceties less provides in favor of completely hard-coded default values.

My intention was to let admins override if they want, but still have default values provided through less/CSS.

If default values are hard-coded, users who copy the sample ini wholesale will need to modify it if colors change.

I'm not against hard-coding it, but this implementation allows overriding if wanted, and otherwise doesn't modify any existing styles.

File: docs/content/doc/advanced/customizing-gitea.en-us.md [hunk 1]
```diff
@@ -165,3 +165,19 @@ To add a custom Readme, add a markdown formatted file (without an `.md` extensio
...
+```ini
+[heatmap]
+# Light theme with a blue heatmap!
+gitea-0 = 0a0a0a
+gitea-1 = 06264f
+gitea-2 = 0f3684
+gitea-3 = 1956ba
+gitea-4 = 2176df
+gitea-5 = 3d9bdf
+```
```
Comment (mrsdizzie) [inline docs/content/doc/advanced/customizing-gitea.en-us.md:189]: Maybe add a second example (dark theme) with a different theme name so it is 100% clear that 'gitea' is the theme name here and not a general value to be used

Comment (jolheiser) [reply docs/content/doc/advanced/customizing-gitea.en-us.md:189]: Done

**Review by mrsdizzie — APPROVED** [review]

**Comment by lafriks** [comment]
Imho this is not needed, this could be overrrided with custom css file, no need to add additional settings

**Comment by jolheiser** [comment]
> Imho this is not needed, this could be overrrided with custom css file, no need to add additional settings

That's a good point (admittedly I never thought of that)  
I suppose then the only advantage is not messing with a `custom/header`.

**Comment by mrsdizzie** [comment]
It could be nice if there was an option to automatically load css/js files that are in a certain folder(s) rather than needing to modifying a template to include them, as you then loose out on any automatic update of that template in the future.

**Comment by jolheiser** [comment]
`custom/header` is actually blank by default, I believe its intended purpose is things like this.  
I completely forgot it existed, though.

**Comment by mrsdizzie** [comment]
ah right I was confusing base/header|footer and also forgot those.

**Comment by 6543** [comment]
> custom/header is actually blank by default, I believe its intended purpose is things like this.
I completely forgot it existed, though.

but it would make sence to add a description to https://docs.gitea.io/en-us/customizing-gitea/
how to ONLY set/update heatmap (with a smal example)

**Pull request closed without merging**